### PR TITLE
[Cody] add callout for scope selector

### PR DESCRIPTION
**Pull request by thenamankumar**
closes: #54007
closes: #54012

- Adds dismissable callout for scope selector
- Add settings page link to embeddings status icon
- Fix bug where chat id in the URL for /cody was getting updated even when the user was logged out or cody was disabled.

image

## Test plan

- visit /cody
- dismiss callout or open scope selector should be dismissed
- reload, should not appear again

- open scope selector
- click on the embeddings status icon, which should open the embeddings settings page in a new tab
- the repo should not be selected/deselected.

### CODE REVIEW

**Comment by almeidapaulooliveira** [comment]
@thenamankumar, a quick fixing for the Callout. Please remove the Cody icon.

Also, a request from @toolmantim, to reduce the paddings, to make all elements smaller.

I can fine-tune the CSS directly on your PR if you think it's helpful. Since you have a lot on your plate.

**Comment by almeidapaulooliveira** [comment]
@thenamankumar, I pushed the icon removal and fine-tuning on CSS to the PR.

**Review by almeidapaulooliveira — APPROVED** [review]

**Comment by almeidapaulooliveira** [comment]
@thenamankumar - The embedding link (on the App) is opening in the browser. It should direct to the settings but in the same window. 

I think on the web, it could open in the same tab as well.

https://github.com/sourcegraph/sourcegraph/assets/64257673/4cce0718-062e-4388-83aa-c27f11137068

**Comment by thenamankumar** [comment]
Thanks for catching it. Fixed @almeidapaulooliveira

**Review by chwarwick — APPROVED** [review]
I tried it out on app and it worked well

File: client/web/src/cody/components/ScopeSelector/RepositoriesSelectorPopover.tsx [hunk 1]
```diff
@@ -484,13 +502,15 @@ const EmbeddingExistsIcon: React.FC<{ repo: { embeddingExists: boolean } }> = Re
...
                 }
             >
-                <Icon
-                    aria-hidden={true}
-                    className={classNames({
-                        [styles.embeddingIconNoEmbeddings]: !embeddingExists,
-                    })}
-                    svgPath={embeddingExists ? mdiDatabaseCheckOutline : mdiDatabaseRemoveOutline}
-                />
+                <Link to="/site-admin/embeddings" className="text-body" onClick={event => event.stopPropagation()}>
```
[the remark this answers is not in the document]
Comment (thenamankumar) [reply client/web/src/cody/components/ScopeSelector/RepositoriesSelectorPopover.tsx:505]: Oh yes, good find. I create a PR soon.

[the remark this answers is not in the document]
Comment (thenamankumar) [reply client/web/src/cody/components/ScopeSelector/RepositoriesSelectorPopover.tsx:505]: fix: https://github.com/sourcegraph/sourcegraph/pull/54090

**Pull request merged**